src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -408,7 +401,56 @@ internal class VmrBackFlower(
 
         await targetRepo.CommitAsync(commitMessage, false, cancellationToken: cancellationToken);
         await targetRepo.ResetWorkingTree();
+        await workBranch.MergeBackAsync(commitMessage);
 
         return true;
     }
+
+    private async Task<(SourceMapping, string)> PrepareVmrAndRepo(
+        string mappingName,
+        ILocalGitRepo targetRepo,
+        string? shaToFlow,
+        string baseBranch,
+        string targetBranch,
+        CancellationToken cancellationToken)
+    {
+        if (shaToFlow is null)
+        {
+            shaToFlow = await _localGitClient.GetShaForRefAsync(_vmrInfo.VmrPath);
+        }
+        else
+        {
+            await _vmrCloneManager.PrepareVmrAsync(shaToFlow, CancellationToken.None);
+        }
+
+        SourceMapping mapping = _dependencyTracker.GetMapping(mappingName);
+        ISourceComponent repoInfo = _sourceManifest.GetRepoVersion(mappingName);
+
+        var remotes = new[] { mapping.DefaultRemote, repoInfo.RemoteUri }
+            .Distinct()
+            .OrderRemotesByLocalPublicOther()
+            .ToArray();
+
+        // Refresh the repo
+        await targetRepo.FetchAllAsync(remotes, cancellationToken);
```

Comment:
`FetchAllAsync` can throw an exception in `AddRemoteIfMissingAsync`:
```
result.ThrowIfFailed($"Failed to add remote {remoteName} ({repoUrl}) to {repoPath}");
```
do we want to `try catch` and handle an exception here or in `FetchAllAsync`?

Reply:
Technically many more exceptions can be thrown from there but I think we want to let them through at the moment. If we see this is problematic, I'd add resiliency later but I'd rather no go too defensive yet.